common-java-libs/org.wso2.apk.apimgt.impl/src/main/java/org/wso2/apk/apimgt/impl/dao/impl/ApiDAOImpl.java

```diff
@@ -3103,26 +3102,34 @@ public DevPortalAPI getDevPortalAPI(Organization organization, String apiId) thr
                 String json = resultSet.getString("ARTIFACT");
                 ObjectMapper mapper = new ObjectMapper();
                 JsonNode tree = mapper.readTree(json);
-                api = mapper.treeToValue(tree, PublisherAPI.class);
+                devPortalAPI = mapper.treeToValue(tree, DevPortalAPI.class);
                 String mediaType = resultSet.getString("MEDIA_TYPE");
                 InputStream apiDefinitionBlob = resultSet.getBinaryStream("API_DEFINITION");
                 if (apiDefinitionBlob != null) {
                     if (StringUtils.equals("swagger.json", mediaType)) {
-                        api.setSwaggerDefinition(APIMgtDBUtil.getStringFromInputStream(apiDefinitionBlob));
-                    } else if (StringUtils.equals("asyncapi.json", mediaType)) {
-                        api.setAsyncApiDefinition(APIMgtDBUtil.getStringFromInputStream(apiDefinitionBlob));
+                        devPortalAPI.setSwaggerDefinition(APIMgtDBUtil.getStringFromInputStream(apiDefinitionBlob));
                     }
-                }
-                API apiObject = APIMapper.INSTANCE.toApi(api);
-                devPortalAPI = APIMapper.INSTANCE.toDevPortalApi(apiObject);
-                if (APIConstants.API_GLOBAL_VISIBILITY.equals(api.getVisibility())) {
+//                    else if (StringUtils.equals("asyncapi.json", mediaType)) {
```

Comment:
If you are keeping these commented lines for future use, lets add todo tag to keep track of them. Or remove them completely if not needed anymore

Reply:
please check on other commented places too on this PR

Reply:
added a todo and resolved